yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,822 @@
+    description "This type defines an OSI NET address using a pattern,
+    An example net-address is 49.0123.6452.1972.00";
+  }
+
+  typedef if-state-type {
+    type enumeration {
+      enum "up" {
+        value 0;
+        configd:help "Up state."; 
+        description "Up state.";
+      }
+      enum "down" {
+        value 1;
+        configd:help "Down state";
+        description "Down state";
+      }
+    }
+    configd:help "This type defines the state of an interface";
+    description "This type defines the state of an interface";
+  }
+
+  typedef adj-state-type {
+    type enumeration {
+      enum "up" {
+        value 0;
+        configd:help "State indicates the adjacency is established.";
+        description "State indicates the adjacency is established.";
+      }
+      enum "down" {
+        value 1;
+        configd:help "State indicates the adjacency is NOT established.";
+        description "State indicates the adjacency is NOT established.";
+      }
+      enum "init" {
+        value 2;
+        configd:help "State indicates the adjacency is establishing.";
+        description "State indicates the adjacency is establishing.";
+      }
+      enum "failed" {
+        value 3;
+        configd:help "State indicates the adjacency is failed.";
+        description "State indicates the adjacency is failed.";
+      }
+    }
+    configd:help "This type defines states of an adjacency";
+    description "This type defines states of an adjacency";
+  }
+
+  typedef metric-style-type {
+    type enumeration {
+      enum "narrow" {
+        value 0;
+        configd:help "This enum describes narrow metric style";
+        description "This enum describes narrow metric style";
+        reference "RFC1195";
+      }
+      enum "wide" {
+        value 1;
+        configd:help "This enum describes wide metric style";
+        description "This enum describes wide metric style";
+        reference "RFC5305";
+      }
+      enum "transition" {
+        value 2;
+        configd:help "This enum describes transition metric style";
+        description "This enum describes transition metric style";
+      }
+    }
+  }
+
+  typedef authenticate-snp-type {
+    type enumeration {
+      enum "none" {
+        value 0;
+        configd:help "No SNP authentication.";
+        description "No SNP authentication.";
+      }
+      enum "send-only" {
+        value 1;
+        configd:help "Send authenticated PDUs but do not check on receiving.";
+        description "Send authenticated PDUs but do not check on receiving.";
+      }
+      enum "validate" {
+        value 3;
+        configd:help "Send authenticated PDUs and check on receiving.";
+        description "Send authenticated PDUs and check on receiving.";
+      }
+    }
+    default "none";
+    configd:help "SNP PDUs authentication.";
+    description "SNP PDUs authentication.";
+  }
+
+  grouping isis-intf-password {
+    list clear {
```

Comment:
I'm not sure `clear` and `md5` should be lists. Does isisd support configuring multiple passwords of the same type?

Reply:
This is Resolved.